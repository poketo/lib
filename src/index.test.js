import poketo from '.';

describe('poketo', () => {
  describe('constructUrl', () => {
    it('returns a site-specific url', () => {
      const tests = [
        {
          input: 'mangakakalot:urami_koi_koi_urami_koi',
          output: 'http://mangakakalot.com/manga/urami_koi_koi_urami_koi',
        },
        {
          input: 'meraki-scans:senryu-girl:5',
          output: 'http://merakiscans.com/senryu-girl/5',
        },
      ];

      tests.forEach(test => {
        expect(poketo.constructUrl(test.input)).toEqual(test.output);
      });
    });

    it('throws a TypeError without arguments', () => {
      expect(() => {
        poketo.constructUrl();
      }).toThrow(TypeError);
    });

    it('throws an InvalidIdError on invalid arguments', () => {
      expect(() => {
        poketo.constructUrl('https://google.com');
      }).toThrow(poketo.InvalidIdError);
    });

    it('throws an UnsupportedSiteError on invalid arguments', () => {
      expect(() => {
        poketo.constructUrl('google:my-series');
      }).toThrow(poketo.UnsupportedSiteError);
    });
  });

  describe('getType', () => {
    it(`returns 'series' for series inputs`, () => {
      const inputs = [
        'https://merakiscans.com/senryu-girl',
        'meraki-scans:senryu-girl',
        'sen-manga:Yotsubato!',
        'https://mangadex.org/manga/420649',
        'https://mangarock.com/manga/mrs-serie-100100972',
        'https://jaiminisbox.com/reader/series/my-hero-academia',
      ];

      inputs.forEach(input => expect(poketo.getType(input)).toEqual('series'));
    });

    it(`returns 'chapter' for chapter inputs`, () => {
      const inputs = [
        'https://merakiscans.com/senryu-girl/5',
        'meraki-scans:senryu-girl:5',
        'sen-manga:Yotsubato!:82',
        'https://mangadex.org/chapter/420649/1',
        'https://mangarock.com/manga/mrs-serie-100100972/chapter/mrs-chapter-100268832',
        'https://jaiminisbox.com/reader/read/my-hero-academia/en/0/191/page/1',
      ];

      inputs.forEach(input => expect(poketo.getType(input)).toEqual('chapter'));
    });

    it('throws errors on invalid arguments', () => {
      const invalidIds = ['hello', 'h8942nrm23cl;.,/,./23,./'];
      const unsupportedIds = [
        'https://google.com/chapter/420649/1',
        'google:my-series',
        'example:series:chapter-5',
      ];

      const getType = input => () => poketo.getType(input);

      invalidIds.forEach(input => {
        expect(getType(input)).toThrow(poketo.InvalidIdError);
      });

      unsupportedIds.forEach(input => {
        expect(getType(input)).toThrow(poketo.UnsupportedSiteError);
      });
    });
  });

  describe('getSeries', () => {
    it('returns a series with metadata', async () => {
      expect.assertions(2);

      const { chapters: a, ...metadataFromUrl } = await poketo.getSeries(
        'https://mangadex.org/manga/13127',
      );
      const { chapters: b, ...metadataFromId } = await poketo.getSeries(
        'mangadex:13127',
      );

      const updatedWhenever = { updatedAt: expect.any(Number) };

      expect(metadataFromUrl).toMatchSnapshot(updatedWhenever);
      expect(metadataFromId).toMatchSnapshot(updatedWhenever);
    });

    it('returns chapters in the correct order', async () => {
      const { chapters } = await poketo.getSeries('mangadex:13108');

      const sortedChapters = chapters
        .reverse()
        .slice(0, 80)
        .reverse();
      const chapterNumbers = sortedChapters.map(c =>
        parseFloat(c.chapterNumber),
      );
      const volumeNumbers = sortedChapters.map(c => parseFloat(c.volumeNumber));

      // Chapter numbers should appear in descending order
      const sortDescending = arr => arr.sort((b, a) => b - a);
      const sortedChapterNumbers = sortDescending(chapterNumbers);
      const sortedVolumeNumbers = sortDescending(volumeNumbers);

      expect(chapterNumbers).toEqual(sortedChapterNumbers);
      expect(volumeNumbers).toEqual(sortedVolumeNumbers);
    });

    it('throws for invalid urls', async () => {
      expect.assertions(1);

      await expect(
        poketo.getSeries('http://helveticascans.com/r/series/non-existent/'),
      ).rejects.toThrow(poketo.NotFoundError);
    });
  });

  describe('getChapter', () => {
    it('returns a chapter', async () => {
      expect.assertions(2);

      await expect(
        poketo.getChapter('http://merakiscans.com/senryu-girl/2'),
      ).resolves.toMatchSnapshot();
      await expect(
        poketo.getChapter('helvetica-scans:talentless-nana:en/1/2'),
      ).resolves.toMatchSnapshot();
    });

    it('throws for invalid urls', async () => {
      expect.assertions(3);

      await expect(poketo.getChapter('banana:dawg')).rejects.toThrow(
        poketo.UnsupportedSiteError,
      );
      await expect(poketo.getChapter('http://google.com')).rejects.toThrow(
        poketo.UnsupportedSiteError,
      );
      await expect(
        poketo.getChapter('http://helveticascans.com/ba/nan/na'),
      ).rejects.toThrow(poketo.InvalidUrlError);
    });
  });
});
