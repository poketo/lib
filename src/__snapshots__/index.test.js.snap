// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`poketo getChapter returns a chapter 1`] = `
Object {
  "id": "meraki-scans:senryu-girl:2",
  "pages": Array [
    Object {
      "id": 0,
      "url": "http://merakiscans.com/wp-content/manga/21/2/00.jpg",
    },
    Object {
      "id": 1,
      "url": "http://merakiscans.com/wp-content/manga/21/2/001.jpg",
    },
    Object {
      "id": 2,
      "url": "http://merakiscans.com/wp-content/manga/21/2/01.jpg",
    },
    Object {
      "id": 3,
      "url": "http://merakiscans.com/wp-content/manga/21/2/02.jpg",
    },
    Object {
      "id": 4,
      "url": "http://merakiscans.com/wp-content/manga/21/2/03.jpg",
    },
    Object {
      "id": 5,
      "url": "http://merakiscans.com/wp-content/manga/21/2/04.jpg",
    },
    Object {
      "id": 6,
      "url": "http://merakiscans.com/wp-content/manga/21/2/05.jpg",
    },
    Object {
      "id": 7,
      "url": "http://merakiscans.com/wp-content/manga/21/2/06.jpg",
    },
    Object {
      "id": 8,
      "url": "http://merakiscans.com/wp-content/manga/21/2/07.jpg",
    },
    Object {
      "id": 9,
      "url": "http://merakiscans.com/wp-content/manga/21/2/08.jpg",
    },
    Object {
      "id": 10,
      "url": "http://merakiscans.com/wp-content/manga/21/2/09.jpg",
    },
    Object {
      "id": 11,
      "url": "http://merakiscans.com/wp-content/manga/21/2/10.jpg",
    },
    Object {
      "id": 12,
      "url": "http://merakiscans.com/wp-content/manga/21/2/11.jpg",
    },
    Object {
      "id": 13,
      "url": "http://merakiscans.com/wp-content/manga/21/2/12.jpg",
    },
  ],
  "url": "http://merakiscans.com/senryu-girl/2",
}
`;

exports[`poketo getChapter returns a chapter 2`] = `
Object {
  "id": "helvetica-scans:talentless-nana:en/1/2",
  "pages": Array [
    Object {
      "height": 900,
      "id": 3801,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0000.png",
      "width": 600,
    },
    Object {
      "height": 1280,
      "id": 3802,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0089 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3803,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0090 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3804,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0091 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3805,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0092 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3806,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0093 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3807,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0094 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3808,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0095 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3809,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0096 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3810,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0097 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3811,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0098 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3812,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0099 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3813,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0100 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3814,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0101 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3815,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0102 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3816,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0103 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3817,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0104 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3818,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0105 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3819,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0106 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3820,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0107 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3821,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0108 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3822,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0109 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3823,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0110 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3824,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0111 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3825,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0112 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3826,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0113 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3827,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0114 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3828,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0115 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3829,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0116 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3830,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0117 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3831,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0118 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3832,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0119 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3833,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0120 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3834,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0121 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3835,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0122 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3836,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0123 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3837,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0124 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3838,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0125 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3839,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0126 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3840,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0127 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3841,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0128 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3842,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0129 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3843,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0130 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3844,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0131 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3845,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0132 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3846,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0133 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3847,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0134 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3848,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0135 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3849,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0136 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3850,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0137 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3851,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0138 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3852,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0139 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3853,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0140 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3854,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0141 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3855,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0142 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3856,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0143 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3857,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0144 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3858,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0145 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3859,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0146 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3860,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0147 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3861,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0148 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3862,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0149 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3863,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0150 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3864,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0151 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3865,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0152 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3866,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0153 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3867,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0154 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3868,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0155 copy.png",
      "width": 900,
    },
    Object {
      "height": 1280,
      "id": 3869,
      "url": "https://helveticascans.com/r/content/comics/talentless-nana_5935ea4001c59/2-0-backtrack_595b173e601f4/0156 copy.png",
      "width": 900,
    },
  ],
  "url": "https://helveticascans.com/r/read/talentless-nana/en/1/2/page/1",
}
`;

exports[`poketo getSeries returns a series with metadata 1`] = `
Object {
  "author": "Akitaka",
  "coverImageUrl": "https://mangadex.org/images/manga/13127.large.jpg?1529770840",
  "description": "The Twelve City, a land governed by the leaders of the families who bear in their names the Chinese zodiac. In a city where youkai and humans coexist, the cat youkai who holds a grudge against the Rat Family appears in front of Nekuni Kyouichi, the head of the Rat Family. It's a malice-covered Hateful cat who is aiming for his life. But that youkai is Nekogasaki Natsuho, who's also extremely anxious around other people... Just like the saying \\"Curiosity killed the cat\\" goes, will these feelings also kill the cat?",
  "id": "mangadex:13127",
  "publicationStatus": "ONGOING",
  "site": Object {
    "id": "mangadex",
    "name": "Mangadex",
  },
  "slug": "13127",
  "supportsReading": true,
  "title": "Uramikoi, Koi, Uramikoi.",
  "updatedAt": Any<Number>,
  "url": "https://mangadex.org/title/13127",
}
`;

exports[`poketo getSeries returns a series with metadata 2`] = `
Object {
  "author": "Akitaka",
  "coverImageUrl": "https://mangadex.org/images/manga/13127.large.jpg?1529770840",
  "description": "The Twelve City, a land governed by the leaders of the families who bear in their names the Chinese zodiac. In a city where youkai and humans coexist, the cat youkai who holds a grudge against the Rat Family appears in front of Nekuni Kyouichi, the head of the Rat Family. It's a malice-covered Hateful cat who is aiming for his life. But that youkai is Nekogasaki Natsuho, who's also extremely anxious around other people... Just like the saying \\"Curiosity killed the cat\\" goes, will these feelings also kill the cat?",
  "id": "mangadex:13127",
  "publicationStatus": "ONGOING",
  "site": Object {
    "id": "mangadex",
    "name": "Mangadex",
  },
  "slug": "13127",
  "supportsReading": true,
  "title": "Uramikoi, Koi, Uramikoi.",
  "updatedAt": Any<Number>,
  "url": "https://mangadex.org/title/13127",
}
`;
